lib/api/BackbeatAPI.js

```diff
@@ -0,0 +1,131 @@
+'use strict'; // eslint-disable-line
+
+const Kafka = require('node-rdkafka');
+const zookeeper = require('node-zookeeper-client');
+
+/**
+ * Class representing Backbeat API endpoints and internals
+ *
+ * @class
+ */
+class BackbeatAPI {
+    /**
+     * @constructor
+     * @param {object} config - configurations for setup
+     * @param {werelogs.Logger} logger - Logger object
+     */
+    constructor(config, logger) {
+        this._zkEndpoint = config.zookeeper.connectionString;
+        this._repConfig = config.extensions.replication;
+        this._kafkaHost = config.kafka.hosts;
+        this._topic = this._repConfig.topic;
+        this._groupId = this._repConfig.queueProcessor.groupId;
+        this._maxISR = config.server.replicas;
+        this._logger = logger;
+
+        // Instantiate Kafka consumer and producer once only instead of
+        // continuously opening and closing connection.
+        this._consumer = new Kafka.KafkaConsumer({
+            'group.id': this._groupId,
+            'metadata.broker.list': this._kafkaHost,
+            'enable.auto.commit': false,
+        });
+        this._consumer.connect();
+
+        this._producer = new Kafka.Producer({
+            'metadata.broker.list': this._kafkaHost,
+            'dr_cb': true,  // delivery report
+        });
+        this._producer.connect();
+
+        this.xproducer = new Kafka.Producer({
+            'metadata.broker.list': this._kafkaHost,
+            'dr_cb': true,  // delivery report
+        });
+
+        this._zkClient = zookeeper.createClient(this._zkEndpoint);
+        this._zkClient.connect();
+    }
+
+    /**
+     * Check if incoming request is valid
+     * @param {string} route - request route
+     * @return {boolean} true/false
+     */
+    isValidRoute(route) {
+        // refactor this
+        const validPaths = ['/_/healthcheck', '/_/throughput'];
+        return validPaths.includes(route);
+    }
+
+    /**
+     * Check if Kafka, Zookeeper, Producer, Consumer are connected
+     * @return {boolean} true/false
+     */
+    isConnected() {
+        // TODO:
+        // Check if Kafka and Zookeeper are connected
+        // Check if any node has fewer ISRs than normal (?)
+        return this._consumer.isConnected() && this._producer.isConnected()
+            && (this._zkClient.getState().name === 'SYNC_CONNECTED');
+    }
+
+    /**
+     * Get Kafka healthcheck
+     * @param {BackbeatRequest} backbeatRequest - Backbeat request object
+     * @param {function} cb - callback for response
+     * @return {undefined}
+     */
+    healthcheck(backbeatRequest, cb) {
+        this._producer.getMetadata({}, (err, meta) => {
+            if (err) {
+                // setup backbeatRequest error properly
+                return cb(backbeatRequest, err);
```

Comment:
Conventionally the `err` should be returned as first argument, and the rest of the args follows.